crates/agglayer-clock/src/block.rs

```diff
@@ -219,34 +262,50 @@ where
                     break;
                 }
                 block_result = Self::recv_block(&mut stream) => {
-                    let block = block_result?;
+                    let block = block_result.map_err(|e| {
+                        error!(error = %e, "Failed to receive block from stream");
+                        agglayer_telemetry::clock::record_connection_failed();
+                        e
+                    })?;
+
                     if block.number <= self.latest_seen_block {
-                        trace!("Skipping block: number={}, latest_seen_block={}", block.number, self.latest_seen_block);
+                        trace!(
+                            block_number = block.number,
+                            latest_seen = self.latest_seen_block,
+                            "Skipping already processed block"
+                        );
                         continue;
                     }
-                    trace!(
-                        "L1 Block received: timestamp={}, number={}, hash={}",
-                        block.timestamp,
-                        block.number,
-                        block.hash
+
+                    debug!(
+                        block_number = block.number,
+                        block_hash = %block.hash,
+                        timestamp = block.timestamp,
+                        blocks_to_process = block.number - self.latest_seen_block,
+                        "Received new L1 block"
                     );
 
                     // Overwrite the block number to simulate an overflow
                     // This is used for testing purposes only and doesn't affect the production
                     // code.
                     fail::fail_point!("block_clock::BlockClock::run::overwrite_block_number_on_new_block");
```

Comment:
Should this go under some test configuration feature?

Reply:
According to the `fail` docs:

> Fail points generation by this macro is disabled by default, and can be enabled where relevant with the `failpoints` Cargo feature.

We enable the feature in `dev-dependencies` only. Larger code blocks that only exist to support fail points are already guarded by `#[cfg(test)]`.

Reply:
Still, I would prefer that it is visually separated from the rest of the code. It is not great to have the test code scattered around the critical production path.